### PR TITLE
Added file type identification for upload source

### PR DESCRIPTION
**Pull request by schaefi**
The azurectl compute storage upload command will now detect the
filetype and apply the decompression only for detected types.
If no supported decompressor type is found the file is uploaded
as it is

### CODE REVIEW

**Comment by schaefi** [comment]
The damn magic implementation we provide does not exist on other distros, I'm going to change the code to just use file that will work. Please do not yet review this

File: doc/man/Makefile [hunk 1]
```diff
@@ -17,5 +17,8 @@ install:
 		install -m 644 $$i ${man_prefix}/man1 ;\
 	done
 
+show:
```
Comment (bear454) [inline doc/man/Makefile]: Is this related?  :stuck_out_tongue:

**Comment by bear454** [comment]
Looks good to me. :+1:

File: doc/man/Makefile (hunk 1 above)
Comment (schaefi) [inline doc/man/Makefile]: not really but I struggled with the man page change to look good and instead to remember the command to render the page I thought it's ok to just add the show target ;)

I still enjoy the early phase of the project which allows us to just do things :)

**Comment by schaefi** [comment]
cool thanks for the review, will merge it
